app/actions/views/account_notifications.js

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) 2017 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import {updateUserNotifyProps} from 'service/actions/users';
+import {Preferences} from 'service/constants';
+import {savePreferences} from 'service/actions/preferences';
+
+export function handleUpdateUserNotifyProps(notifyProps) {
+    return async (dispatch, getState) => {
+        const state = getState();
+        const config = state.entities.general.config;
+
+        let email = notifyProps.email;
+        if (config.EnableEmailBatching === 'true' && email !== 'false') {
+            const emailInterval = [{
+                user_id: notifyProps.user_id,
+                category: Preferences.CATEGORY_NOTIFICATIONS,
+                name: Preferences.EMAIL_INTERVAL,
+                value: email
```

Comment:
This doesn't look right. The email interval should be a number, but a string representation of a boolean